js/stores/windowStore.js

```diff
@@ -898,14 +898,18 @@ const doAction = (action) => {
       } else {
         doAction({actionType: windowConstants.WINDOW_SET_CONTEXT_MENU_DETAIL})
       }
-      doAction({actionType: windowConstants.WINDOW_SET_SUBMENU_SELECTED_INDEX})
+      doAction({actionType: windowConstants.WINDOW_SET_MENUBAR_SELECTED_INDEX})
+      doAction({actionType: windowConstants.WINDOW_SET_CONTEXT_MENU_SELECTED_INDEX})
       doAction({actionType: windowConstants.WINDOW_SET_BOOKMARKS_TOOLBAR_SELECTED_FOLDER_ID})
       break
-    case windowConstants.WINDOW_SET_SUBMENU_SELECTED_INDEX:
-      windowState = windowState.setIn(['ui', 'menubar', 'selectedIndex'],
-        Array.isArray(action.index)
-        ? action.index
-        : null)
+    case windowConstants.WINDOW_SET_MENUBAR_SELECTED_INDEX:
+      windowState = windowState.setIn(['ui', 'menubar', 'selectedIndex'], action.index)
+      break
+    case windowConstants.WINDOW_SET_CONTEXT_MENU_SELECTED_INDEX:
+      windowState = windowState.setIn(['ui', 'contextMenu', 'selectedIndex'],
+          Array.isArray(action.index)
+          ? action.index
+          : null)
```

Comment:
~I believe you have the logic for these two items mixed up; since `WINDOW_SET_SUBMENU_SELECTED_INDEX` was changed to `WINDOW_SET_MENUBAR_SELECTED_INDEX`, the array check should be there (which then also has a default to null).  You may also keep the check in for `WINDOW_SET_CONTEXT_MENU_SELECTED_INDEX`, but instead defaulting to `[0]` like you did above in `main.js`~

Disregard my comment above (striked out).  I now see that MenubarSelectedIndex is always having an integer value; If you'd like to keep type checking in place, you can instead maybe compare it against typeof 'number' and make sure it's greater than 0

Regarding the check for `WINDOW_SET_CONTEXT_MENU_SELECTED_INDEX`, you may want to default to `[0]` like you do above in `main.js` (instead of null)